client/handle_http.go

```diff
@@ -2122,31 +2126,52 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 }
 
 // Walk a remote directory in a WebDAV server, emitting the files discovered
-func (te *TransferEngine) walkDirDownload(job *clientTransferJob, transfers []transferAttemptDetails, files chan *clientTransferFile, url *url.URL) error {
-	// Create the client to walk the filesystem
-	rootUrl := *url
-	if job.job.namespace.DirListHost != "" {
-		// Parse the dir list host
-		dirListURL, err := url.Parse(job.job.namespace.DirListHost)
+func (te *TransferEngine) walkDirDownload(job *clientTransferJob, transfers []transferAttemptDetails, files chan *clientTransferFile, remoteURL *url.URL) error {
+	// Query the director a PROPFIND to see if we can get our directory listing
+	resp, err := queryDirector("PROPFIND", remoteURL.Path, job.job.directorUrl+"/api/v1.0/director/origin")
+	if err != nil {
+		return err
+	}
+
+	collectionsUrl := *remoteURL
+	// If we get a "method not allowed", we go back to the old method
+	if resp.StatusCode == http.StatusMethodNotAllowed {
+		log.Debugln("PROPFIND against director failed, trying to get collections url from namespace")
```

Comment:
Since this is at the client, our error messages will be much more user-facing than they would be in a Pelican server like the Director. You should make the final decision, but I think we could come up with an error message that's a little more user friendly -- I don't think the average person knows what PROPFIND is or what a collections URL means.